### PR TITLE
Add touch support for NavigationControl's compass button + add pitch indication to icon

### PR DESCRIPTION
**Pull request by pakastin**
Related to #3405

NavigationControl's compass button was missing touch drag support. Added that.

You can try the functionality here: https://aviamaps.com/map

### CODE REVIEW

**Comment by pakastin** [comment]
Tapping broke, will fix that now

File: src/ui/handler/drag_rotate.js [hunk 1]
```diff
@@ -258,6 +270,8 @@ class DragRotateHandler {
     _unbind() {
         window.document.removeEventListener('mousemove', this._onMouseMove, {capture: true});
         window.document.removeEventListener('mouseup', this._onMouseUp);
+        window.document.removeEventListener('touchmove', this._onMouseMove, { capture: true });
```
Comment (cs09g) [inline src/ui/handler/drag_rotate.js]: should be followed the coding convention `{ capture: true }` to  `{capture: true}` as line no.271

**Comment by pakastin** [comment]
Tapping bug now fixed, so everything is working as expected!

File: src/ui/handler/drag_rotate.js (hunk 1 above)
Comment (pakastin) [reply src/ui/handler/drag_rotate.js]: Didn't change coding convention it already had..

**Comment by pakastin** [comment]
Linting issues now fixed!

File: src/ui/handler/drag_rotate.js (hunk 1 above)
Comment (cs09g) [reply src/ui/handler/drag_rotate.js]: I see, there's already used in mixed way.

Comment (pakastin) [reply src/ui/handler/drag_rotate.js]: However I fixed that ;)

**Comment by pakastin** [comment]
Bonus feature: compass icon now pitches when pitch is changed

**Comment by pakastin** [comment]
> The second feature is also discussing on #8208. I think one PR should be covered one feature.
 
that was the plan, but accidentally came into this PR.. 😕

**Comment by cs09g** [comment]
How could you skip the issue already registered. I'd like you to divide this PR for each feature.

**Comment by andrewharvey** [comment]
> How could you skip the issue already registered. I'd like you to divide this PR for each feature.

@pakastin I think it's okay for both these features to be done in this single PR given they are related, and the touch support is fairly minor. I don't think it's worth the overhead of effort separating them, unless you want to, or unless the maintainers ask...

**Comment by pakastin** [comment]
Ok, perfect! They're both basically patches IMO..

File: src/ui/control/navigation_control.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ class NavigationControl {
     }
 
     _rotateCompassArrow() {
-        const rotate = `rotate(${this._map.transform.angle * (180 / Math.PI)}deg)`;
+        const rotate = `rotateX(${this._map.transform._pitch * (180 / Math.PI)}deg) rotateZ(${this._map.transform.angle * (180 / Math.PI)}deg)`;
```
Comment (asheemmamoowala) [inline src/ui/control/navigation_control.js]: It would be better to use the `transform#pitch` property instead of the internal variable.

Comment (pakastin) [reply src/ui/control/navigation_control.js]: Ok, fixed now! 👍

**Comment by pakastin** [comment]
Any news on merging this?

**Comment by pakastin** [comment]
Good, I can update to v1.0.0 then 😎

**Pull request merged**